os/drivers/lcd/lcd_dev.c

```diff
@@ -0,0 +1,395 @@
+            int row;
+
+            for (row = lcd_area->row_start; row <= lcd_area->row_end; row++)
+              {
+                ret = priv->planeinfo[lcd_area->planeno].putrun(priv->lcd_ptr, row,
+                                             lcd_area->col_start, buf,
+                                             cols);
+                if (ret < 0)
+                  {
+                    break;
+                  }
+
+                buf += row_size;
+              }
+          }
+      }
+      break;
+    case LCDDEVIO_GETPOWER:
+      {
+        *((FAR int *)arg) = priv->lcd_ptr->getpower(priv->lcd_ptr, (int)arg);
+      }
+      break;
+    case LCDDEVIO_SETPOWER:
+      {
+        ret = priv->lcd_ptr->setpower(priv->lcd_ptr, (int)arg);
+      }
+      break;
+    case LCDDEVIO_GETCONTRAST:
+      {
+        *((FAR int *)arg) = priv->lcd_ptr->getcontrast(priv->lcd_ptr);
+      }
+      break;
+    case LCDDEVIO_SETCONTRAST:
+      {
+        ret = priv->lcd_ptr->setcontrast(priv->lcd_ptr, (unsigned int)arg);
+      }
+      break;
+    case LCDDEVIO_GETPLANEINFO:
+      {
+        *((FAR struct lcd_planeinfo_s *)arg) = priv->planeinfo[0];
+      }
+      break;
+    case LCDDEVIO_GETVIDEOINFO:
+      {
+        ret = priv->lcd_ptr->getvideoinfo(priv->lcd_ptr,
+                                          (FAR struct fb_videoinfo_s *)arg);
+      }
+      break;
+    case LCDDEVIO_INIT:
+      {
+          priv->lcd_ptr->init(priv->lcd_ptr);
+      }
+      break;
+    case LCDDEVIO_CLEAR:
+      {
+          priv->lcd_ptr->clear(priv->lcd_ptr);
+      }
+      break;
+    default:
+      {
+        if (priv->lcd_ptr->ioctl)
+          {
+            ret = priv->lcd_ptr->ioctl(priv->lcd_ptr, cmd, arg);
+          }
+        else
+          {
+            printf("ERROR: Unsupported IOCTL command: %d\n", cmd);
+            ret = -ENOTTY;
+          }
+      }
+      break;
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: lcddev_register
+ *
+ * Description:
+ *   Register the LCD character driver as /dev/lcdN.
+ *
+ * Input Parameters:
+ *   devno - The LCD device number.
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success.  Otherwise a negated errno value is
+ *   returned to indicate the nature of the failure.
+ *
+ ****************************************************************************/
+
+int lcddev_register(int devno)
+{
+  FAR struct lcddev_dev_s *priv;
+  int ret = OK;
+  char devname[16] = {0,};;
+
+  /* Allocate a new lcd_dev driver instance */
+
+  priv = (struct lcddev_dev_s *)kmm_zalloc(sizeof(struct lcddev_dev_s));
+
+  if (!priv)
+    {
+      return -ENOMEM;
+    }
+
+  //pthread_mutex_init(&priv->lock, NULL);
+
+  priv->lcd_ptr = board_lcd_getdev(devno);
```

Comment:
so it need to be removed here. it should be hided from board specific code. because lcd_dev is generic virtual driver